gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java

```diff
@@ -280,17 +227,41 @@ public TagWrapper(int wrapDepth, String preWrap, String postWrap) {
         return (GQuery) o;
       }
       if (o instanceof Function) {
-        return $((Function) o);
-      }
-      if (JsUtils.isElement(o)) {
-        return $(JsUtils.<Element> cast(o));
-      }
-      if (o instanceof JavaScriptObject) {
-        return $((JavaScriptObject) o);
+        return $(((Function) o).getElement());
```

Comment:
But this method doesn't exist anymore, aren't you introducing a loop here ? Idem for the other call of `$()` inside this method